docs: Clarify _meta property

domdomegg opened the pull request:
## Summary
- Remove confusing "reserved by MCP" language from `_meta` property documentation
- Clearly specify what keys are actually reserved: `progressToken` (backwards compatibility) and keys with `modelcontextprotocol.*` or `mcp.*` prefixes
- Explicitly state that all other keys are available for arbitrary use by implementations
- Update both draft and 2025-06-18 specification versions

This resolves the documentation ambiguity raised in the issue where "reserved" was used in conflicting ways.

Fixes #264

domdomegg committed:
Clarify _meta property documentation

Remove confusing "reserved by MCP" language and clearly specify what keys are actually reserved:
- progressToken (for backwards compatibility)
- Keys with modelcontextprotocol.* or mcp.* prefixes

All other keys are available for arbitrary use by implementations.

Fixes #264

domdomegg committed:
Update generated schema.json file

domdomegg committed:
Fix documentation formatting with prettier

domdomegg committed:
Fix more CI things :/

localden commented:
This feels like it _materially_ changes the interpretation of `_meta` so typically would need a SEP, but also I'd be curious if this actually impacts any servers or clients directly with the change (i.e., are there any constructs where `_meta` is treated as reservered today where this can introduce potential ambiguities)?

domdomegg committed:
Update _meta documentation to use reverse DNS and only reserve io.modelcontextprotocol prefix

- Use reverse DNS notation (io.modelcontextprotocol instead of modelcontextprotocol.io)
- Only reserve io.modelcontextprotocol prefix (removed mcp.* reservation)
- Add SHOULD recommendation for implementations to use reverse DNS
- Update examples to demonstrate reverse DNS pattern
- Only modify draft specification (revert 2025-06-18 changes)

Aligns with SEP proposal for _meta clarification.

domdomegg committed:
Merge origin/main into adamj/clarify-meta-documentation

Resolved conflicts by:
- Keeping both _meta Reserved Keys documentation and icons documentation in index.mdx
- Using origin/main's progressToken comment in schema.ts
- Regenerating schema.json and schema.mdx from the corrected schema.ts
